### PR TITLE
[KOGITO-9428]Update Operator SDK from 1.25 to 1.31

### PR DESCRIPTION
**Pull request by desmax74**
See: https://issues.redhat.com/browse/KOGITO-9428

and related:
https://github.com/operator-framework/operator-sdk/issues/6285 
https://github.com/operator-framework/operator-sdk/pull/6419

**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: Makefile
Comment (ricardozanini) [inline Makefile:265]: lol

File: hack/checkCreatedAtAnnotation.sh
Comment (ricardozanini) [inline hack/checkCreatedAtAnnotation.sh:1]: Can we keep this file name lowercase, or use dashes instead? The other files in this directory follow this pattern.

**Review by ricardozanini — COMMENTED** [review]
It seems simple enough 🤔 
We can merge this after 1.41.x release since we might cherrypick/fix bugs for TP1 up until there. It will be easier for us if the SDK version remains the same for this work.

File: hack/checkCreatedAtAnnotation.sh
Comment (desmax74) [reply hack/checkCreatedAtAnnotation.sh:1]: Ok

**Review by davidesalerno — APPROVED** [review]

**Comment by ricardozanini** [comment]
Please hold until #185

File: hack/check-created-at-annotation.sh
Comment (desmax74) [reply hack/check-created-at-annotation.sh:23]: yep

File: hack/check-created-at-annotation.sh
Comment (radtriste) [reply hack/check-created-at-annotation.sh:22]: Could you add a small comment on what it does ? (apart from the issues above)

**Review by ricardozanini — APPROVED** [review]

**Comment by spolti** [comment]
can we hold until we get https://github.com/kiegroup/kogito-serverless-operator/pull/239 merged?

**Comment by domhanak** [comment]
@ricardozanini looks stale, how to proceed with this PR?

**Comment by ricardozanini** [comment]
> @ricardozanini looks stale, how to proceed with this PR?

This is relevant, but I'm holding off until we settle on the features being introduced at the moment. There are many requests for features atm and upgrading the base SDK may lead to many errors since we have many collaborators atm.

**Comment by ricardozanini** [comment]
The code base changed a lot from this last iteration. @desmax74 do you think you can rebase? Otherwise, I'll upgrade to the latest in a new PR.

**Pull request closed without merging**
